pkg/controllers/dynakube/extension/eec/statefulset.go

```diff
@@ -113,6 +124,33 @@ func (r *reconciler) createOrUpdateStatefulset(ctx context.Context) error {
 	return nil
 }
 
+func (r *reconciler) buildTemplateAnnotations(ctx context.Context) (map[string]string, error) {
+	templateAnnotations := map[string]string{}
+
+	if r.dk.Spec.Templates.ExtensionExecutionController.Annotations != nil {
+		maps.Copy(templateAnnotations, r.dk.Spec.Templates.ExtensionExecutionController.Annotations)
```

Comment:
why not just use:
```
templateAnnotations = r.dk.Spec.Templates.ExtensionExecutionController.Annotations
```
?

Reply:
`r.dk.Spec.Templates.ExtensionExecutionController.Annotations` is nil if no custom annotations are set

Reply:
but you check it on line 130, or am I miss something?

Reply:
If I do
`templateAnnotations = r.dk.Spec.Templates.ExtensionExecutionController.Annotations`
Wouldn't it be copying by reference? Meaning, if I update templateAnnotations afterwards, I'm updating the CRD value

Reply:
nope,
```
package main

import "fmt"

func main() {
	m := map[string]string{}
	var anotherMap map[string]string
	fmt.Println(m)

	if anotherMap != nil {
		m = anotherMap
		fmt.Println(m)
	}

	m["test"] = "test"

	fmt.Println(anotherMap)

}
```
>map[]
> map[]

>Program exited.

Reply:
You are right :+1: thanks for testing it!
[Fixed](https://github.com/Dynatrace/dynatrace-operator/pull/3843/commits/d359c469dfc8d438e88a6e9228f2a760bc4e806e)